src/main/java/com/conveyal/datatools/manager/jobs/DeployJob.java

```diff
@@ -563,6 +573,12 @@ public void jobFinished () {
         Persistence.deployments.replace(deployment.id, deployment);
         // Send notification to those subscribed to updates for the deployment.
         NotifyUsersForSubscriptionJob.createNotification("deployment-updated", deployment.id, message);
+        if (!status.error) {
+            // A new FetchSingleFeedJob could be started after the AutoDeployJob has made sure no fetching jobs exist and
+            // the DeployJob has started. Therefore this new feed version wouldn't result in a new DeployJob getting kicked
+            // off since there was already one running. To catch this new feed version another auto deploy job is started.
+            DataManager.heavyExecutor.execute(new AutoDeployJob(deployment.parentProject(), owner));
+        }
```

Comment:
Something that just came to mind is that this could result in an endless loop of deployments. So if a deployment is initialized, then the deploy job finishes successfully there would be no stopping point within either this job or the AutoDeploy job until an error occurs.